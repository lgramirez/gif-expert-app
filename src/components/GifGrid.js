import React, { useEffect, useState } from "react";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url =
      "https://api.giphy.com/v1/gifs/search?q=rick+and+morthy&limit=10&api_key=G3x1V4VNTyP4snSskhcKfu11XFgJbUSw";
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images.downsized_medium.url,
      };
    });

    console.log(gifs);
    setImages(gifs);
  };

  return (
    <>
      <h3>{category}</h3>

      {images.map((image) => {
        return <GifGridItem key={image.id} {...image} />;
      })}
    </>
  );
};
