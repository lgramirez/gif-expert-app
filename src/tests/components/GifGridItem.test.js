import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";
import "../../setupTests";

describe("Pruebas para GifGridItem", () => {
  const title = "titulo";
  const url = "https://localhost/algo.jpg";

  test("debe mostar el componente correctamente (hacer match con un snapshot)", () => {
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    expect(wrapper).toMatchSnapshot();
  });
});
