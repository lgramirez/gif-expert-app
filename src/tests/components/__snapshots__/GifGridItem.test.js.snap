// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para GifGridItem debe mostar el componente correctamente (hacer match con un snapshot) 1`] = `
<div
  className="card animate__animated animate__fadeInRight"
>
  <img
    alt="titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
